domains/transfer-agreements/TransferAgreements.API/API/ApiModels/Responses/TransferAgreementHistoryEntryDto.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using API.Data;
+
+namespace API.ApiModels.Responses;
+
+public record TransferAgreementHistoryEntryDto(
```

Comment:
All the properties here are on the same level, but they fall logically in two groups: 

1) Properties about the change (e.g. the ChangeAction and then history/audit date)
2) Properties for the value of the TransferAgreement

I suggest to make this distinction clear. This can be done by having a Property dedicated to group 2